src/Avalonia.Controls.ColorPicker/Helpers/ColorPickerHelpers.cs

```diff
@@ -619,45 +616,17 @@ Color GetColor(double componentValue)
         /// <param name="pixelWidth">The pixel width of the bitmap.</param>
         /// <param name="pixelHeight">The pixel height of the bitmap.</param>
         /// <returns>A new <see cref="WriteableBitmap"/>.</returns>
-        public static WriteableBitmap CreateBitmapFromPixelData(
-            ArrayList<byte> bgraPixelData,
+        public static unsafe Bitmap CreateBitmapFromPixelData(
+            PooledList<byte> bgraPixelData,
             int pixelWidth,
             int pixelHeight)
         {
-            // Standard may need to change on some devices
-            Vector dpi = new Vector(96, 96);
-
-            var bitmap = new WriteableBitmap(
-                new PixelSize(pixelWidth, pixelHeight),
-                dpi,
-                PixelFormat.Bgra8888,
-                AlphaFormat.Premul);
-
-            using (var frameBuffer = bitmap.Lock())
+            fixed (byte* array = bgraPixelData.Span)
```

Comment:
I am fundamentally against using unsafe code at the control-level of Avalonia. NO unsafe/native code access should be required for controls like this. Therefore, I am against this change on those grounds. However, if the Avalonia team wants to override I'll live with it.

I would prefer WriteableBitmap work like all the other XAML frameworks and use that here.

That said, there is a clear performance benefit just pointing a Bitmap to an existing array of bytes. Your code now has zero-copy (original code was 2 copies, then with array list 1 copy). Now you are just using the original array as constructed for the bitmap so zero copies.

Reply:
I understand that. We could potentially add Bitmap ctor overload accepting Span<byte>. But it won't really add any value, as with skia we would end up pinning this span from being GCd either way.

> Now you are just using the original array as constructed for the bitmap so zero copies.

Even better - original array is pooled, for the most part it won't be re-allocated even for many color pickers.

Reply:
Yea, I've nothing against the performance gains here. They are needed for some of the platforms (WASM, Android) anyway. So let's agree this is a temporary improvement until we can do something with composition brushes and GPU acceleration.